shared/src/androidMain/kotlin/dev/johnoreilly/confetti/test/screenshot/ScreenshotTestRule.kt

```diff
@@ -0,0 +1,118 @@
+package dev.johnoreilly.confetti.test.screenshot
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.view.View
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalView
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onRoot
+import com.quickbird.snapshot.Diffing
+import com.quickbird.snapshot.Snapshotting
+import com.quickbird.snapshot.fileSnapshotting
+import com.quickbird.snapshot.snapshot
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.rules.RuleChain
+import org.junit.rules.TestName
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+
+fun createScreenshotTestRule(): ScreenshotTestRule {
+    return ScreenshotTestRuleImpl()
+}
+
+interface ScreenshotTestRule: TestRule {
+    /**
+     * Set configuration properties important for the screenshot testing rule.
+     *
+     * @param tolerance This is the percentage difference allowed for the test to pass.
+     * @param snapshotTransformer
+     * @param record Set this to true if you want to record the test
+     */
+    fun setConfiguration(
+        tolerance: Float = 0.1f,
+        snapshotTransformer: SnapshotTransformer = SnapshotTransformer.None,
+        record: Boolean = false
+    )
```

Comment:
Dont know if this is a good idea to be honest 🤔 at least not to have the record there. Maybe that could be in the constructor instead